### PR TITLE
[Test] Add test to validate crash log creation

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
run log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-C5Y358/

File: tests/cephadm/test_verify_crash_reports.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+def run(ceph_cluster, **kw):
+    """Verify redeploy for a specific service
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    # Check for mgr nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Identify the osd pid
+    cmd = "pgrep osd"
```
Comment (vamahaja) [inline tests/cephadm/test_verify_crash_reports.py:17]: can't we use - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py#L599 ? If no then create generic method and add ^^^

File: tests/cephadm/test_verify_crash_reports.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    out, _ = installer.exec_command(cmd=cmd, sudo=True)
+
+    if not out:
+        raise OperationFailedError("No osd daemons running")
+    pid_to_kill = out.split("\n")[0].strip()
+
+    # Introduce a crash in OSD daemon
+    cmd = f"kill -s SIGSEGV {pid_to_kill}"
```
Comment (vamahaja) [inline tests/cephadm/test_verify_crash_reports.py:25]: Same as above

File: tests/cephadm/test_verify_crash_reports.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    crash_stat = CephAdm(installer).ceph.crash.stat()
+    log.info(f"Crash stats after stopping container are '{crash_stat}'")
+
+    # Validate if the crash is posted in /var/lib/ceph/<fsid>/crash, /var/lib/ceph/<fsid>/crash/posted directories.
+    fsid = CephAdm(installer).ceph.fsid()
+
+    folders = [f"/var/lib/ceph/{fsid}/crash", f"/var/lib/ceph/{fsid}/crash/posted"]
+    log_found = False
+    for folder in folders:
+        cmd = f"ls {folder}"
```
Comment (vamahaja) [inline tests/cephadm/test_verify_crash_reports.py:38]: Use - https://github.com/red-hat-storage/cephci/blob/master/ceph/ceph.py#L2074

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
@vamahaja all review comments addressed

updated run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-Q7VECJ/

File: cli/utilities/utils.py [hunk 4]
```diff
@@ -800,3 +800,29 @@ def copy_ceph_sshkey_to_host(installer, node, user="root", key=CEPH_PUB_KEY):
...
+def kill_process(node, pid_to_kill):
+    """
+    Kills the process using process id
+    Args:
+        node (ceph): Ceph node
+        pid_to_kill (str/list): Pid to kill
+    """
+    pids = pid_to_kill if isinstance(pid_to_kill, list) else [pid_to_kill]
+    for pid in pids:
+        cmd = f"kill -s SIGSEGV {pid}"
```
Comment (vamahaja) [inline cli/utilities/utils.py:814]: Why separate command for each pid ? Can't we pass list of pids ?

File: tests/cephadm/test_verify_crash_reports.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    log.info(f"Crash stats after stopping container are '{crash_stat}'")
+
+    # Validate if the crash is posted in /var/lib/ceph/<fsid>/crash, /var/lib/ceph/<fsid>/crash/posted directories.
+    folders = [f"/var/lib/ceph/{fsid}/crash", f"/var/lib/ceph/{fsid}/crash/posted"]
+    log_found = False
+    for folder in folders:
+        dirs = installer.get_dir_list(dir_path=folder, sudo=True)
+        if dirs:
+            log_found = True
+    if not log_found:
```
Comment (vamahaja) [inline tests/cephadm/test_verify_crash_reports.py:37]: no need to have this logic, test case should fail even one directory fails to generate crash logs.

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Review by vamahaja — APPROVED** [review]

**Pull request merged**